math/calculator.cpp

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright 2020 @author David Leal
+ * @file
+ *
+ * Basic calculator: sum, subtract, multiply, and divide.
+ */
+
+#include <iostream>
+#include <cstdlib>
+
+/**
+ * Same as the number typed by the user, but it's saved
+ * in the new node so it won't be re-writed.
+ */
+
+struct Node {
+    float datatype;
+};
+
+/** 
+ * Prototypes
+ */
+
+void requestData(Node*);
+void pause();
+
+/**
+ * --------------
+ * Main function
+ * --------------
+ */
+
+int main() {
+    Node* new_node = new Node();
+    requestData(new_node);
+
+    /**
+     * Release allocated memory used in `new_node`.
+     */
+
+    delete new_node;
+
+    pause();
+    return 0;
+}
+
+/**
+ * Similar as system("pause"), but secure.
+ */
+
+void pause() {
+    std::cout << "Press ENTER key to continue . . . ";
+
+    std::cin.get();
```

Comment:
only one would be enough

Reply:
Tested with Visual Studio 2019: it doesn't work with one.

Reply:
[this is the reason why](https://stackoverflow.com/questions/10656224/using-cin-get-twice)
Essentially, the newline character is not getting flushed and is retained in the `istream` until the first `can.get` is invoked.

Reply:
Will use `rewind(stdin)`, as `fflush(stdin)` doesn't work (at least for me).
If requested, I'll use `fflush(stdin)`.